### PR TITLE
Fix admin-ui redirect

### PR DESCRIPTION
**Pull request by ericbottard**

### CODE REVIEW

File: spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/AdminApplication.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package org.springframework.cloud.data.rest;
```
Comment (markfisher) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/AdminApplication.java]: We do not want to move the AdminApplication up. It belongs under 'rest' (the root of spring-cloud-data-rest). Why is this necessary?

File: spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/ui/UiController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+/**
+ * A simple controller to trigger a redirect to the index page of the admin ui (which comes packaged as a dependency).
+ *
+ * @author Eric Bottard
+ */
+// Note how this controller is NOT a RestController
```
Comment (markfisher) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/ui/UiController.java]: should include an explanation of *why* (and does this belong in actual JavaDoc?)

File: spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/AdminApplication.java (hunk 1 above)
Comment (ericbottard) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/AdminApplication.java]: Because the "UI" is not REST _per se_.

Comment (markfisher) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/AdminApplication.java]: It is simply a redirect to the REST API right? The more important thing here is that the "-rest" suffix is mapping to the ".rest" package as the "namespace" of this project.

Comment (ericbottard) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/AdminApplication.java]: Hmm, not sure what you're saying. The "ui" part is not a REST-API. It's a regular `.html` app, that is _consuming_ the API.
I understand what you're saying wrt namespacing. IMO, there are two solutions:
* go back to having the `ui` serving part inside `.rest` and live with it
* create an extra project, which is the real "admin", made of 2 parts: the REST endpoints and the ui

Comment (markfisher) [inline spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/AdminApplication.java]: Almost, but I wasn't thinking of creating another project. Rather we could rename `spring-cloud-data-rest` to `spring-cloud-data-admin`. Then the top-level package for all code in the project would be: `org.springframework.cloud.data.admin`. The following sub-packages would still exist: config, controller, repository. Only `AdminAppplication` would reside directly under the `.admin` top level package. The `UiController` would sit alongside the "REST" Controllers within the `controller` subpackage (REST being considered an implementation detail of all the other Controllers).

**Comment by ericbottard** [comment]
Reworked by renaming spring-cloud-data-rest to spring-cloud-data-admin as discussed. Also removed some redundant `@Import` and `@ComponentScan`.

File: spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/repository/package-info.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Spring Cloud Data REST repository classes.
```
Comment (markfisher) [inline spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/repository/package-info.java]: s/REST/Admin/

File: spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/controller/package-info.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Spring Cloud Data REST & UI Controllers.
```
Comment (markfisher) [inline spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/controller/package-info.java]: "Spring Cloud Data *Admin* REST & UI Controllers." (or we can probably leave out "REST & UI").

File: spring-cloud-data-admin/src/test/java/org/springframework/cloud/data/admin/controller/TaskControllerTests.java [hunk 5]
```diff
@@ -29,10 +29,12 @@
 import org.junit.runner.RunWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.data.admin.AdminApplication;
 import org.springframework.cloud.data.core.TaskDefinition;
```
Comment (markfisher) [inline spring-cloud-data-admin/src/test/java/org/springframework/cloud/data/admin/controller/TaskControllerTests.java]: looks like these imports did not get re-arranged in alphabetical order

File: spring-cloud-data-admin/src/test/java/org/springframework/cloud/data/admin/repository/InMemoryTaskDefinitionRepositoryTests.java [hunk 6]
```diff
@@ -29,6 +29,8 @@
 import org.junit.Before;
 import org.junit.Test;
 
+import org.springframework.cloud.data.admin.repository.DuplicateTaskException;
```
Comment (markfisher) [inline spring-cloud-data-admin/src/test/java/org/springframework/cloud/data/admin/repository/InMemoryTaskDefinitionRepositoryTests.java]: these 2 imports should not be necessary (same package)

File: spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/config/CloudConfiguration.java [hunk 7]
```diff
@@ -14,21 +14,23 @@
...
 
 /**
- * Configuration for cloud profiles (cloud, lattice)
+ * Configuration for cloud profiles (cloud, lattice).
  *
  * @author Mark Fisher
  * @author Thomas Risberg
  */
 @Profile("cloud")
+@Configuration
```
Comment (markfisher) [inline spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/config/CloudConfiguration.java]: minor thing but should we have `@Profile` on the line after `@Configuration` since this class is primarily Configuration but has its applicability narrowed by the Profile specified? Also, I believe that would be consistent with other cases within the project.

**Comment by markfisher** [comment]
Affected READMEs (top-level and spring-cloud-data-admin) should be updated as part of the PR.

**Comment by markfisher** [comment]
will address comments while merging
